core/src/main/java/com/linecorp/armeria/server/RedirectService.java

```diff
@@ -0,0 +1,185 @@
+ * sb.service("/old/{var}", new RedirectService(HttpStatus.MOVED_PERMANENTLY, "/new/{var}");
+ * sb.service("/old/{var}", new RedirectService("http://user:name@localhost:8080/search/{var}");
+ * }</pre>
+ */
+public class RedirectService extends AbstractHttpService {
+
+    private static final Pattern VALID_DEFAULT_URI_PATTERN = Pattern
+            .compile("(?:(?:^https?:/{2}(?:([^:]+:)?[^:@]+@)?[^:]+)(?::[0-9]{1,5})?)?" +
+                    "(?:/[^/{}:]+|/:[^/{}]+|/\\{[^/{}]+})+/?");
+    private static final Pattern PATTERN_PARAMS_START = Pattern.compile("/:|/\\{");
+
+    private final HttpStatus httpStatus;
+    private final Function<? super ServiceRequestContext, String> locationFunction;
+    private Set<String> paramNames;
+
+    /**
+     * Creates a new Redirect {@link Service} that redirects to the location constructed with the specified
+     * {@code locationPattern}.
+     *
+     * @param locationPattern a {@link PathMapping} string that the bound {@link Service} will redirect to.
+     *
+     * @throws IllegalArgumentException if the specified {@code locationPattern} is unsupported or invalid
+     */
+    public RedirectService(String locationPattern) {
+        this(HttpStatus.TEMPORARY_REDIRECT, locationPattern);
+    }
+
+    /**
+     * Creates a new Redirect {@link Service} that redirects to the location returned by
+     * {@code locationFunction}.
+     *
+     * @param locationFunction a {@link Function} that takes a {@link ServiceRequestContext}
+     *      and returns new location path.
+     */
+    public RedirectService(Function<? super ServiceRequestContext, String> locationFunction) {
+        this(HttpStatus.TEMPORARY_REDIRECT, locationFunction);
+    }
+
+    /**
+     * Creates a new Redirect {@link Service} that redirects to the location constructed with the specified
+     * {@code locationPattern}.
+     *
+     * @param redirectStatus the {@link HttpStatus} that the {@link Service} will return.
+     * @param locationPattern a {@link PathMapping} string that the bound {@link Service} will redirect to.
+     *
+     * @throws IllegalArgumentException if the specified {@code locationPattern} is unsupported or invalid
+     */
+    public RedirectService(HttpStatus redirectStatus, String locationPattern) {
+        this(redirectStatus, toLocationFunction(locationPattern));
+
+        Matcher m = PATTERN_PARAMS_START.matcher(locationPattern);
+        if (m.find()) {
+            paramNames = PathMapping.of(locationPattern.substring(m.start())).paramNames();
+        }
+    }
+
+    /**
+     * Creates a new Redirect {@link Service} that redirects to the location returned by
+     * {@code locationFunction}.
+     *
+     * @param redirectStatus the {@link HttpStatus} that the {@link Service} will return.
+     * @param locationFunction a {@link Function} that takes a {@link ServiceRequestContext}
+     *      and returns new location path.
+     */
+    public RedirectService(HttpStatus redirectStatus,
```

Comment:
nit: Formatting. This constructor would be formatted with our formatting rule as follows:
```
    public RedirectService(HttpStatus redirectStatus,
                           Function<? super ServiceRequestContext, String> locationFunction) {
        requireNonNull(redirectStatus, "redirectStatus");
        requireNonNull(locationFunction, "locationFunction");
        if (redirectStatus.compareTo(HttpStatus.MULTIPLE_CHOICES) < 0 ||
            redirectStatus.compareTo(HttpStatus.TEMPORARY_REDIRECT) > 0) {
            throw new IllegalArgumentException("redirectStatus: " + redirectStatus + " (expected: 300 .. 307)");
        }
        httpStatus = redirectStatus;
        this.locationFunction = locationFunction;
    }
```
Would you check [CONTRIBUTING.md](https://github.com/line/armeria/blob/master/CONTRIBUTING.md) about formatting?

Reply:
@hyangtack Referred to other classes and have changed accordingly, thoroughly i hope.
However, there's no mention of Constructor/function param and multi conditional indentation, maybe we should add it?